app.js

```diff
@@ -1,64 +1,70 @@
-let bookLibrary = {};
-let jsonBooks = JSON.parse(localStorage.getItem('books'));
+const form = document.querySelector('form');
+const bookSection = document.querySelector('#library');
 
-function addBook(tittleName, authorName) {
-  bookLibrary[tittleName] = {
-    tittle: tittleName,
-    author: authorName,
-  };
-  const jsonBooks = JSON.stringify(bookLibrary);
-  localStorage.setItem('books', jsonBooks);
-}
+let jsonBooks = JSON.parse(localStorage.getItem('booksStorage'));
```

Comment:
When I load the page for the first time, the `jsonBooks` variable is undefined and so I get an error resulting from an attempt to call the map function. To fix the issue, you can initialize an empty array when the key wasn't found in the local storage.
e.g
```javascript
let jsonBooks = JSON.parse(localStorage.getItem('booksStorage')) || [];
```
![image](https://github.com/pulgosodog/awesome-books/assets/20554583/d649ae95-15ba-498d-a91a-ec5145be7628)

Reply:
Fixed! I moved `jsonBooks = JSON.parse(localStorage.getItem('booksStorage'));` and `bookLibrary = jsonBooks.map((book) => new Books(book.title, book.author));` to the DOMContentLoaded event, and set the `jsonBooks` when it is declared to an emty array.

Reply:
Appreciate the feedback! 🙌